### PR TITLE
EnHy + objects docs

### PR DESCRIPTION
**Pull request by hensldm**
Documents EnHy as well as the different objects surrounding it (except os_anime which is large and tbh I didn't feel like documenting all the animtions). Regarding the objects I decided to go with the naming style `gHylian` + attributes, where attributes describe the npc (old/young, man/woman, details describing their head, etc.) . Any suggestions are welcome of course.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Hy/z_en_hy.h [hunk 1]
```diff
@@ -40,25 +63,25 @@ typedef struct EnHy {
...
     /* 0x0214 */ s8 waypoint;
-    /* 0x0215 */ s8 unk_215;
+    /* 0x0215 */ s8 playedSfx;
     /* 0x0216 */ char unk_216[2]; // unused
     /* 0x0218 */ s16 curEyeIndex;
     /* 0x021A */ s16 nextEyeIndexTimer;
-    /* 0x021C */ s16 unk_21C[16]; // bodyWiggleY ?
-    /* 0x023C */ s16 unk_23C[16]; // bodyWiggleZ ?
+    /* 0x021C */ s16 fidgetTableY[16]; // bodyWiggleY ?
+    /* 0x023C */ s16 fidgetTableZ[16]; // bodyWiggleZ ?
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Hy/z_en_hy.h:78]: Im not 100% sure about calling this a "fidget table" in general. I am aware that mm just recently moved to this terminology, but I wasnt around at the time to take part in the conversation.

Guess its ok for now, but will prob want to discuss it on a broader scale at some point in the future.
Can prob remove the the comments on them too

Comment (hensldm) [reply src/overlays/actors/ovl_En_Hy/z_en_hy.h]: I'm always down to discuss names, so if we do have other suggestions please just let me know. Removed the comments at least.

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Hy/z_en_hy.c [hunk 2]
```diff
@@ -1188,46 +1431,47 @@ void EnHy_Draw(Actor* thisx, PlayState* play) {
...
-            // ENHY_TYPE_BOB_18
-            case ENHY_TYPE_BJI_19:
-            case ENHY_TYPE_AHG_20:
+        envColorSeg8 = sModelInfo[ENHY_GET_TYPE(&this->actor)].envColorSeg8;
+        envColorSeg9 = sModelInfo[ENHY_GET_TYPE(&this->actor)].envColorSeg9;
+
+        switch (ENHY_GET_TYPE(&this->actor)) {
+            // ENHY_TYPE_DOG_LADY
+            // ENHY_TYPE_WOMAN_3
+            case ENHY_TYPE_MAN_1_BEARD:
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Hy/z_en_hy.c:1443]: Why are some of these types commented in the switch? If their case isn't here, then the code below wouldn't run?

Comment (hensldm) [reply src/overlays/actors/ovl_En_Hy/z_en_hy.c:1443]: Yeah the code doesn't run for those cases. The comments were just there from the original decomp I think. I should probably just remove them.

Comment (Dragorn421) [reply src/overlays/actors/ovl_En_Hy/z_en_hy.c]: I put those comments in, to highlight which types are *not* there (they are placed in the enum's order)
but yeah feel free to remove :shrug:

Comment (engineer124) [reply src/overlays/actors/ovl_En_Hy/z_en_hy.c]: I see, could always put the comments in the default case?

Comment (hensldm) [reply src/overlays/actors/ovl_En_Hy/z_en_hy.c]: Sure why not.

**Review by cadmic — APPROVED** [review]

**Pull request merged**